### PR TITLE
Allow configuration of the subnet the LB is placed in

### PR DESCRIPTION
**Pull request by malt3**
This is an optional parameter that can either be left uninitialized to keep the old behavior or be set as a helm value or per LB.

We at edgeless systems have a use-case for yawol where we need the loadbalancer to be in the same network but a different subnet (cidr) than the kubernetes nodes.
This is a requirement for the network encryption we are using in our clusters.
Basically, we need to enforce that any traffic in the node subnet cidr belongs to a k8s node (and can thus enforce it to be encrypted).
Loadbalancer traffic must therefore be part of a separate subnet.

### CODE REVIEW

**Comment by dergeberl** [comment]
Thanks for your contribution 🥳 

Can you please add a test to check that `createPort` is correct if there is a subnet set and if there is no subnet set?

**Comment by malt3** [comment]
> Can you please add a test to check that `createPort` is correct if there is a subnet set and if there is no subnet set?

I added tests for the loadbalancermachine controller behavior. Please let me know if there is more testing needed!

**Comment by dergeberl** [comment]
> I added tests for the loadbalancermachine controller behavior. Please let me know if there is more testing needed!

Can you please add one more test for the yawol-cloud-controller. To test that it also works that a subnet can be set via an annotation. 
Like this:
https://github.com/stackitcloud/yawol/blob/19dcae4f7bee3a8f41275e26e1f0220fcfb8885e/controllers/yawol-cloud-controller/targetcontroller/service_controller_test.go#L2187

File: controllers/yawol-controller/loadbalancermachine/loadbalancermachine_controller_test.go [hunk 1]
```diff
@@ -240,6 +240,60 @@ var _ = Describe("load balancer machine", Serial, Ordered, func() {
...
+
+				g.Expect(actual.Status.ServerID).ToNot(BeNil())
+				g.Expect(actual.Status.DefaultPortID).ToNot(BeNil())
+
+				_, err := client.ServerClientObj.Get(ctx, *actual.Status.ServerID)
+				g.Expect(err).To(Succeed())
+
+				port, err := client.PortClientObj.Get(ctx, *actual.Status.DefaultPortID)
+				g.Expect(err).To(Succeed())
+				g.Expect(len(port.FixedIPs)).To(Equal(1))
```
Comment (Kumm-Kai) [inline controllers/yawol-controller/loadbalancermachine/loadbalancermachine_controller_test.go:263]: Use `.To(HaveLen(1))` instead.

File: controllers/yawol-controller/loadbalancer/loadbalancer_controller.go
Comment (Kumm-Kai) [inline controllers/yawol-controller/loadbalancer/loadbalancer_controller.go:1]: Can you please also extend the tests in `loadbalancer_controller_test.go`

**Review by Kumm-Kai — CHANGES_REQUESTED** [review]
Thanks for your contribution, just a few additional comments :)

File: controllers/yawol-controller/loadbalancer/loadbalancer_controller.go
Comment (malt3) [reply controllers/yawol-controller/loadbalancer/loadbalancer_controller.go:1]: I'm not sure what exactly you would like to validate here. This file also does not contain tests for the main `DefaultNetwork.NetworkID` field.

Comment (Kumm-Kai) [reply controllers/yawol-controller/loadbalancer/loadbalancer_controller.go:1]: Tbh I'm also not sure what we've wanted to have covered there. Maybe to ensure that if `DefaultNetwork.SubnetID` is set that the port will use this subnet. But I think we can keep it like it is for now.

**Review by Kumm-Kai — APPROVED** [review]

**Comment by Kumm-Kai** [comment]
Thanks for your contribution! ❤️

**Pull request merged**